src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailResources.groovy

```diff
@@ -101,4 +101,15 @@ public class TrailsResource extends Resource {
 			zipCode, smallDrop,largeDrop, woodRide, skinny, largeJump,smallJump, gap)
 		ok(trailResult(trails)).build()
 	}
+
+	@GET
+    @Path ('/{id: \\d+}')
+	Response getByID (@PathParam('id') Integer id) {
+		Trail trail = trailDAO.getTrailByID(id)
+		if (trail) {
+			return ok(trailResource(trail)).build()
```

Comment:
Up to you, but these return statements in this `if else` are not necessary since these would be the last lines executed in this method.